test/functional/webauthn_credentials_controller_test.rb

```diff
@@ -171,6 +171,68 @@ class WebauthnCredentialsControllerTest < ActionController::TestCase
 
       should respond_with :unprocessable_entity
     end
+
+    context "when otp is not yet registered" do
```

Comment:
Is this the same case as `when correctly verifying a challenge`?

Also, there's a lot of duplicate code in each setup block, we can create a context with `when signed in` and have a setup block with creating a user, signing in and setting the origin and client.